setup.py

```diff
@@ -780,76 +826,135 @@ def get_provider_package_from_package_id(package_id: str):
-    add_provider_packages_to_requirements("all_dbs", ALL_DB_PROVIDERS)
-    add_provider_packages_to_requirements("devel_hadoop", ["apache.hdfs", "apache.hive", "presto"])
+        add_provider_packages_to_extras_requirements(provider, [provider])
+    add_provider_packages_to_extras_requirements("all", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_ci", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_all", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("all_dbs", ALL_DB_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_hadoop", ["apache.hdfs", "apache.hive", "presto"])
+
+
+class Develop(develop_orig):
+    """Forces removal of providers in editable mode."""
+
+    def run(self):
+        self.announce('Installing in editable mode. Uninstalling provider packages!', level=log.INFO)
+        # We need to run "python3 -m pip" because it might be that older PIP binary is in the path
+        # And it results with an error when running pip directly (cannot import pip module)
+        # also PIP does not have a stable API so we have to run subprocesses ¯\_(ツ)_/¯
+        try:
+            installed_packages = (
+                subprocess.check_output(["python3", "-m", "pip", "freeze"]).decode().splitlines()
+            )
+            airflow_provider_packages = [
+                package_line.split("=")[0]
+                for package_line in installed_packages
+                if package_line.startswith("apache-airflow-providers")
+            ]
+            self.announce(f'Uninstalling ${airflow_provider_packages}!', level=log.INFO)
+            subprocess.check_call(["python3", "-m", "pip", "uninstall", "--yes", *airflow_provider_packages])
+        except subprocess.CalledProcessError as e:
+            self.announce(f'Error when uninstalling airflow provider packages: {e}!', level=log.WARN)
+        super().run()
+
+
+class Install(install_orig):
+    """Forces installation of providers from sources in editable mode."""
 
+    def run(self):
+        self.announce('Standard installation. Providers are installed from packages', level=log.INFO)
+        super().run()
 
-def do_setup():
-    """Perform the Airflow package setup."""
+
+def do_setup() -> None:
+    """
+    Perform the Airflow package setup.
+    Most values come from setup.cfg, only the dynamically calculated ones are passed to setup
+    function call. See https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html
+    """
     setup_kwargs = {}
 
-    if os.getenv('INSTALL_PROVIDERS_FROM_SOURCES') == 'true':
-        # Only specify this if we need this option, otherwise let default from
-        # setup.cfg control this (kwargs in setup() call take priority)
-        setup_kwargs['packages'] = find_namespace_packages(include=['airflow*'])
+    def include_provider_namespace_packages_when_installing_from_sources() -> None:
```

Comment:
Why is this a separate function?

Reply:
The main reason was that I wanted to  name it appropriately. I think everywhere in a code (especially in such case where you do not care about a performance, if you have a need to write a paragraph of comment, it's much better to replace it withe a function that have the comment converted to pydoc, and is named appropriately, even if the code is short (but not obvious why we are doing it).

This way when you look at the code you can focus on what the function does (fucntion name), why we do it (pydoc when for example hovering your mouse over the function) and how it does it (function body) separately and understand it at different levels.